run-tests.php

```diff
@@ -2971,24 +2972,31 @@ function settings2params(array $ini_settings): string
     $settings = '';
 
     foreach ($ini_settings as $name => $value) {
```

Comment:
for people reviewing this: the old loop was much more complex than necessary, and even wrong, for unix. I don't think the windows-loop is correct either, but I don't have a Windows computer to test on, and I'm not comfortable editing the windows-side of the loop. for Unix, all of this stuff can, and was, replaced with literally 3 lines of code 👍

Reply:
>TBF it has even more branches now

yeah but the new branch is super simple (3 lines) and correct, while the old branches were complex-ish and incorrect (incorrect for unix at least)

>We could at least apply the same optimization to Windows (handling values and arrays the same)

yeah, but then we need to test it. I have no easy way of testing it. If someone else wants to fix it, that's fine, but I'd rather not touch the windows branch. 

>and applying the " to ' fix in both cases. I don't understand what it's for 

the existing logic suggest that removing `"` is only a special-case for lists/arrays on Windows, not applicable to plain strings and not applicable to unix, and it looks like it's a hack in the first place, i don't think we should add the `"` thing to plain strings without a deeper understanding of why it exists. I don't understand it either.

Reply:
If someone want to change/optimize/fix the Windows branch, that's fine, but it shouldn't be me. I don't even have a Windows system to test on.

Reply:
@iluuu1994 i think the whole function could be written as
```php
function settings2params(array $ini_settings): string
{
    $settings = '';

    foreach ($ini_settings as $name => $value) {
        if (is_array($value)) {
            $deep = true;
        } else {
            $deep = false;
            $value = [$value];
        }
        foreach ($value as $val) {
            if (IS_WINDOWS && $deep && str_starts_with($val, '"') && str_ends_with($val, '"')) {
                $val = substr($val, 1, -1);
            }
            $settings .= " -d " . escapeshellarg($name) . "=" . escapeshellarg($val);
        }
        return $settings;
    }
    ```
    but i don't know, and i have no easy way of testing it, and if im wrong, i have to push it to github, then wait 2 hours for github actions tell me if it worked or not, and repeat, until i get it right. it can take hours and its super annoying, and it would be much easier for some guy with an actual Windows system to test on.

Reply:
@iluuu1994 i tried: 8461490203da88c045ca6af3a390b111bd215ffb
turns out using escapeshellarg() on windows breaks random tests due to bugs in escapeshellarg() itself, for example on Windows
```php
escapeshellarg("!E_ERROR");
```
returns
```
"E_ERROR"
```
due to a bug/limitation in escapeshellarg() on windows, and that in turns breaks the test https://github.com/php/php-src/blob/0de8bc270db6c2ffbe34a18393a74b996a010f50/tests/func/011.phpt#L4 (among others)

but fixing escapeshellarg() on windows is out of scope for this PR <.< I'm reverting 8461490203da88c045ca6af3a390b111bd215ffb for now

Reply:
@divinity76 Thanks for checking, separate branches are fine for now. I'll merge this tomorrow.

Reply:
> 

hi
I have a windows computer and would like to help. what needs to be done?

Reply:
@divinity76 escapeshellcmd do the trick if correctly understand problem
var_dump(escapeshellcmd('E_ALL & E_NOTICE | E_PARSE ^ E_DEPRECATED & ~E_WARNING | !E_ERROR'));
will ouput
E_ALL ^& E_NOTICE ^| E_PARSE ^^ E_DEPRECATED ^& ^~E_WARNING ^| ^!E_ERROR

Reply:
@highlyprofessionalscum cool, try it! but then you need to add the quotes yourself, ie
```
var_dump('"' . escapeshellcmd('E_ALL & E_NOTICE | E_PARSE ^ E_DEPRECATED & ~E_WARNING | !E_ERROR') . '"');
```
(not sure if double-quotes `"` or single-quotes `'` works best for Windows, that's another thing you can check yourself)

remember `escapeshellarg("foo bar");`  gives a single argument `'foo bar'`  while `escapeshellcmd('foo bar');` gives 2x arguments `foo` and `bar` :)

Reply:
@highlyprofessionalscum @iluuu1994 come to think of it, i think i would prefer to get this PR merged as-is, then @highlyprofessionalscum can make his own dedicated PR to fix up settings2params-for-windows

Reply:
btw I think escapeshellarg()-on-windows is a shitshow, mostly thanks to Microsoft making it extremely difficult to get right, to the point where nobody really does, including PHP's escapeshellarg(): https://docs.microsoft.com/en-us/archive/blogs/twistylittlepassagesallalike/everyone-quotes-command-line-arguments-the-wrong-way

Reply:
@divinity76 i think we can remove all escaping. test run by proc_open and the docs says 

array|string $command – 
Execute a command and open file pointers for input/output
As of PHP 7.4.0, cmd may be passed as array of command parameters. In this case the process will be opened directly (without going through a shell) and PHP will take care of any necessary argument escaping.

Reply:
@divinity76 
dimli@LAPTOP-KUHG2256 MINGW64 /c/OpenServer2/domains/ts/sf/public (release/1.45)
$  /c/OpenServer2/modules/php/PHP_7.4/php.exe index.php
!!exclamation_marks!!int(0)

Reply:
cmd too loking good
![изображение](https://github.com/php/php-src/assets/49132044/1279923f-0b59-4e03-8e5a-704cce95ea34)

Reply:
@highlyprofessionalscum neat! if you want to update the function, and the code relying on the function, to use arrays instead of strings, that would be better improvement for sure. 

But i think it should be done in it's own dedicated PR, not this one ^^